extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -474,6 +474,38 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
       return null
     }
 
+  /**
+   * Whether to generate ids for the geojson features. When enabled, the `feature.id` property will be auto
+   * assigned based on its index in the `features` array, over-writing any previous values.
+   */
+  val generateId: Boolean?
```

Comment:
is naming something we could change?
imho sounds a bit confusing (not clear it's a boolean)

same applies to the java doc here - I'd rather state clearly we need this value be `true` if feature state API will be used, current documentation is not that clear (at least for me)

Reply:
no, this is defined in style spec, and the documentation also comes from style spec. And actually the description is very informative if you start to digging more about what it actually does.